source/math/FloatMatrix.ooc

```diff
@@ -381,9 +380,12 @@ FloatMatrix: cover {
 		submatrix
 	}
 	inverse: func -> This {
-		version(safe)
-			raise(!this take() isSquare, "Matrix must be square in FloatMatrix inverse")
-		result := this take() cofactors() transpose() / this determinant()
+		t := this take()
+		version(safe) {
+			raise(!t isSquare, "Matrix must be square in FloatMatrix inverse")
+			raise(t determinant() equals(0.f), "Matrix is singular in FloatMatrix inverse")
```

Comment:
Won't this cause extra penalty (albeit minimal) on the non-safe version? Do we really care about optimizing the safe version that much?